WordPress/Classes/ViewRelated/Site Creation/Web Address/DesignSystem/Components/DSButton.swift

```diff
@@ -0,0 +1,165 @@
+        title: String,
+        style: DSButtonStyle,
+        isLoading: Binding<Bool> = .constant(false),
+        action: @escaping () -> Void
+    ) {
+        self._isLoading = isLoading
+        self.action = action
+        self.title = title
+        self.style = style
+    }
+
+    public var body: some View {
+        switch style.size {
+        case .large:
+            button
+        case .medium, .small:
+            button
+                .fixedSize(horizontal: true, vertical: false)
+        }
+    }
+
+    private var button: some View {
+        Button {
+            action()
+            UIImpactFeedbackGenerator(style: .medium).impactOccurred()
+        } label: {
+            if style.emphasis == .tertiary {
+                buttonContent
+            } else {
+                buttonContent
+                    .contentShape(
+                        RoundedRectangle(
+                            cornerRadius: Length.Radius.small
+                        )
+                    )
+            }
+        }
+        .buttonStyle(ScalingButtonStyle(style: style))
+        .disabled(isLoading)
+    }
+
+    private var buttonContent: some View {
+        ZStack {
+            buttonBackground
+            if isLoading {
+                ProgressView()
+                    .tint(Color.white)
+            } else {
+                buttonText
+                    .foregroundStyle(
+                        style.foregroundColor
+                            .opacity(foregroundOpacity)
+                    )
+                    .padding(
+                        .horizontal,
+                        style.size == .small
+                        ? Length.Padding.split
+                        : Length.Padding.medium
+                    )
+            }
+        }
+        .frame(
+            height: style.size == .small
+            ? Length.Padding.large
+            : Length.Padding.max
+        )
+    }
+
+    private var buttonText: some View {
+        switch style.size {
+        case .large:
+            switch style.emphasis {
+            case .primary, .secondary:
+                return Text(title)
+                    .style(.bodyLarge(.emphasized))
+            case .tertiary:
+                return Text(title)
+                    .style(.bodyLarge(.regular))
+            }
+        case .medium:
+            switch style.emphasis {
+            case .primary, .secondary:
+                return Text(title)
+                    .style(.bodyMedium(.emphasized))
+            case .tertiary:
+                return Text(title)
+                    .style(.bodyMedium(.regular))
+            }
+        case .small:
+            switch style.emphasis {
+            case .primary, .secondary:
+                return Text(title)
+                    .style(.bodySmall(.emphasized))
+            case .tertiary:
+                return Text(title)
+                    .style(.bodySmall(.regular))
+            }
+        }
```

Comment:
You can use the enum case as initialiser to flatten the switch-case statements.

```suggestion
        let textStyle: (TextStyle.Weight) -> TextStyle
        let weight: TextStyle.Weight

        switch style.size {
        case .large:
            textStyle = TextStyle.bodyLarge
        case .medium:
            textStyle = TextStyle.bodyMedium
        case .small:
            textStyle = TextStyle.bodySmall
        }

        switch style.emphasis {
        case .primary, .secondary:
            weight = .emphasized
        case .tertiary:
            weight = .regular
        }

        return Text(title).style(textStyle(weight))
```